omod/src/main/java/org/openmrs/module/smartonfhir/util/SmartCachingHolder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.smartonfhir.util;
+
+import java.util.concurrent.TimeUnit;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+public class SmartCachingHolder {
```

Comment:
Can we come up with a better name for this? E.g., `SmartSessionCache`? Do we really need a class for this or can this be integrated into the servlet? Also, if this cache is tying a constant to a single value, how do we deal with more than 1 simultaneous session.

Ideally what I'd like to see is a small class something like:

```java
@Data
public class SmartSession {
    private String visitUuid;
    private String patientUuid;
}
```

And have the cache key between the launch value and an instance of `SmartSession`

Reply:
I'll change the name, thanks

Yes, you are right. It won't work correctly with more than 1 simultaneous session. we can use launch uuid to store the value for every single session.

Reply:
Perfect!